src/SignalR/clients/cpp/src/signalrclient/hub_connection_impl.cpp

```diff
@@ -50,10 +50,8 @@ namespace signalr
 
     void hub_connection_impl::initialize()
     {
-        auto this_hub_connection = shared_from_this();
-
         // weak_ptr prevents a circular dependency leading to memory leak and other problems
-        auto weak_hub_connection = std::weak_ptr<hub_connection_impl>(this_hub_connection);
```

Comment:
Was this calling the copy ctor before?

Reply:
Nvm. I see that shared_from_this() returns a shared_ptr. Now it makes sense why you switched from auto to writing out the full type.